crates/uv/src/commands/project/init.rs

```diff
@@ -0,0 +1,174 @@
+use std::fmt::Write;
+use std::path::PathBuf;
+
+use anyhow::Result;
+
+use owo_colors::OwoColorize;
+use pep508_rs::PackageName;
+use uv_cache::Cache;
+use uv_client::{BaseClientBuilder, Connectivity};
+use uv_configuration::PreviewMode;
+use uv_distribution::pyproject_mut::PyProjectTomlMut;
+use uv_distribution::{ProjectWorkspace, WorkspaceError};
+use uv_fs::Simplified;
+use uv_python::{
+    EnvironmentPreference, PythonFetch, PythonInstallation, PythonPreference, PythonRequest,
+};
+use uv_warnings::warn_user_once;
+
+use crate::commands::ExitStatus;
+use crate::printer::Printer;
+
+/// Add one or more packages to the project requirements.
+#[allow(clippy::single_match_else)]
+pub(crate) async fn init(
+    path: Option<String>,
+    name: Option<PackageName>,
+    no_readme: bool,
+    no_pin: bool,
+    python: Option<String>,
+    python_preference: PythonPreference,
+    python_fetch: PythonFetch,
+    preview: PreviewMode,
+    connectivity: Connectivity,
+    native_tls: bool,
+    cache: &Cache,
+    printer: Printer,
+) -> Result<ExitStatus> {
+    if preview.is_disabled() {
+        warn_user_once!("`uv init` is experimental and may change without warning.");
+    }
+
+    // Discover the current workspace, if it exists.
+    let current_dir = std::env::current_dir()?.canonicalize()?;
+    let workspace = match ProjectWorkspace::discover(&current_dir, None).await {
+        Ok(project) => Some(project),
+        Err(WorkspaceError::MissingPyprojectToml) => None,
+        Err(err) => return Err(err.into()),
+    };
+
+    let project_dir = match path {
+        None => current_dir.clone(),
+        Some(ref path) => PathBuf::from(path),
+    };
+
+    let name = match name {
+        Some(name) => name,
+        None => {
+            // Get the name of the directory.
+            let name = project_dir
+                .file_name()
+                .and_then(|path| path.to_str())
+                .expect("Invalid package name");
+            PackageName::new(name.to_string())?
+        }
+    };
+
+    // Make sure the package does not already exist.
+    if project_dir.join("pyproject.toml").exists() {
+        anyhow::bail!("Package is already initialized")
+    }
+
+    // Create the directory for the project.
+    let src_dir = project_dir.join("src").join(name.as_ref());
+    fs_err::create_dir_all(&src_dir)?;
+
+    // Create the `pyproject.toml`.
+    fs_err::write(
+        project_dir.join("pyproject.toml"),
+        indoc::formatdoc! {r#"
+        [project]
+        name = "{name}"
+        version = "0.1.0"
+        description = "Add your description here"
+        dependencies = []
+        readme = "README.md"
```

Comment:
Ideally only fill in the readme key if actually creating the README. (Rye has a nice goal that it creates a working configuration without extra steps.)